### ISSUE TITLE
Move 'cache' directory to user directory

### ISSUE THREAD

**Issue by tpill90** [tpill90/steam-lancache-prefill#181]
![image](https://user-images.githubusercontent.com/766513/196272687-5b0d73ae-90ba-4bc1-96ab-bf4ef7ff6dc5.png)

**Comment by tpill90** [comment]
```
namespace SteamPrefill.Settings
{
    public static class AppConfig
    {
        static AppConfig()
        {
            (CacheDir, ConfigDir) = GetBaseDirectories();
            
            // Create required folders
            Directory.CreateDirectory(ConfigDir);
            Directory.CreateDirectory(CacheDir);

            AccountSettingsStorePath = Path.Combine(ConfigDir, "account.config");
            UserSelectedAppsPath = Path.Combine(ConfigDir, "selectedAppsToPrefill.json");
        }

        #if DEBUG

        public static bool EnableSteamKitDebugLogs => false;

        public static bool SkipDownloads = false;

        #endif

        public static string SteamCdnUrl => "lancache.steamcontent.com";

        /// <summary>
        /// Increment when there is a breaking change made to the files in the cache directory
        /// </summary>
        private const string CacheDirVersion = "v3";

        private const string ApplicationFolderName = nameof(SteamPrefill);

        /// <summary>
        /// Get the base directories for configuration and cache folders
        /// </summary>
        /// <returns></returns>
        /// <exception cref="NotSupportedException"></exception>
        private static (string cacheDir, string configDir) GetBaseDirectories()
        {
            if (System.OperatingSystem.IsWindows())
            {
                return GetBaseDirectories_Windows();
            }
            
            if (System.OperatingSystem.IsLinux())
            {
                return GetBaseDirectories_XDG();
            }
            
            if (System.OperatingSystem.IsMacOS())
            {
                return GetBaseDirectories_MacOS();
            }
            
            throw new NotSupportedException($"Unknown platform {RuntimeInformation.OSDescription}");
        }

        /// <summary>
        /// Get base directories for Windows
        /// </summary>
        /// <returns></returns>
        private static (string cacheDir, string configDir) GetBaseDirectories_Windows()
        {
            string pathAppData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);

            string pathSteamPrefill = Path.Combine(pathAppData, ApplicationFolderName); 
            return (
                Path.Combine(pathSteamPrefill, "Cache", CacheDirVersion),
                Path.Combine(pathSteamPrefill, "Config")
            );
        }

        /// <summary>
        /// Get base directories for the XDG Base Directory specification (Linux)
        /// </summary>
        /// <returns></returns>
        /// <exception cref="ArgumentOutOfRangeException"></exception>
        private static (string cacheDir, string configDir) GetBaseDirectories_XDG()
        {
            string pathHome =
                Environment.GetEnvironmentVariable("HOME")
                ?? throw new ArgumentNullException("HOME", "Could not determine HOME directory");
            string pathXdgCacheHome =
                Environment.GetEnvironmentVariable("XDG_CACHE_HOME")
                ?? Path.Combine(pathHome, ".cache");
            string pathXdgConfigHome =
                Environment.GetEnvironmentVariable("XDG_CONFIG_HOME")
                ?? Path.Combine(pathHome, ".config");

            return (
                Path.Combine(pathXdgCacheHome, ApplicationFolderName, CacheDirVersion),
                Path.Combine(pathXdgConfigHome, ApplicationFolderName)
            );
        }

        /// <summary>
        /// Get base directories for macOS
        /// </summary>
        /// <returns></returns>
        private static (string cacheDir, string configDir) GetBaseDirectories_MacOS()
        {
            string pathLibraryCaches = Path.GetFullPath("~/Library/Caches");
            string pathLibraryApplicationSupport = Path.GetFullPath("~/Library");
            
            return (
                Path.Combine(pathLibraryCaches, ApplicationFolderName, CacheDirVersion),
                Path.Combine(pathLibraryApplicationSupport, ApplicationFolderName)
            );
        }

        /// <summary>
        /// Downloaded manifests, as well as other metadata, are saved into this directory to speedup future prefill runs.
        /// All data in here should be able to be deleted safely.
        /// </summary>
        public static readonly string CacheDir;

        /// <summary>
        /// Contains user configuration.  Should not be deleted, doing so will reset the app back to defaults.
        /// </summary>
        public static readonly string ConfigDir;

        public static readonly string AccountSettingsStorePath;
        public static readonly string UserSelectedAppsPath;
    }
}
```

**Comment by tpill90** [comment]
Released in v1.7.0 https://github.com/tpill90/steam-lancache-prefill/releases/tag/v1.7.0
